theano/gpuarray/dnn.py

```diff
@@ -418,13 +446,28 @@ def c_headers(self):
         return ['cudnn.h', 'cudnn_helper.h']
 
     def c_header_dirs(self):
-        return [os.path.dirname(__file__), config.dnn.include_path]
+        dirs = [os.path.dirname(__file__)]
+        if config.dnn.include_path:
+            dirs.append(config.dnn.include_path)
+        if config.cuda.include_path:
+            dirs.append(config.cuda.include_path)
+        return dirs
 
     def c_libraries(self):
         return ['cudnn']
 
     def c_lib_dirs(self):
-        return [config.dnn.library_path]
+        if config.dnn.library_path:
+            return [config.dnn.library_path]
+        return []
```

Comment:
I think we should allow an empty string and filter it elsewhere instead of in each op. It make building ops easier I think.

Idem for the other c_* method.